addons/mail/static/src/messaging/entity/messaging/messaging.js

```diff
@@ -53,24 +53,24 @@ function MessagingFactory({ Entity }) {
         async redirect({ id, model }) {
             if (model === 'mail.channel') {
                 const channel = this.env.entities.Thread.channelFromId(id);
-                if (!channel) {
+                if (!channel || !channel.isPinned) {
```

Comment:
Shouldn't we add a test covering that?

Reply:
You mean, the redirect method ? Should be done in this branch you think ?

Reply:
If there was a change needed here, that means there was a bug, so there should be a test covering it, because that seems tricky. I would tend to agree with you, it would be better to add that test in another branch (and the fix too).

Reply:
Don't remember if I added this as a result of a crash in a test or just from looking at the code:
mail manager assumed that if it knows channel, then current user is subscribed to channel.
I think this worked because mail manager was designed like that (and so had lots of limitations as a result).
With the store, we lifted the limitations by adding channel whenever we are aware of them, whether it comes from server data from a message or followers, etc...
That means current user is subscribed to channel only if it knows the channel **and** it is flagged as subscribed.
At the moment, distinction between "pinned" and "subscribed" doesn't exist, so pinned = subscribed. Would be nice to improve that later :)

Reply:
https://www.odoo.com/web#id=2244279&action=3206&active_id=1519&model=project.task&view_type=form&cids=&menu_id=4720